### DISCUSSION TITLE
What do you do when you have a problem using 'if' option after generating a random number?

### DISCUSSION THREAD

**Discussion by asdklfjsad** [community/community#21423] [Other Feature Feedback, Questions, & Ideas]
<pre><code>    #include &lt;stdio.h&gt;
    #include &lt;stdlib.h&gt;
    #include &lt;time.h&gt;

int main()
{
  int a;
  scanf("%d", &amp;a);
  srand(time(NULL));
  int random1 = rand() % ((90-0)+1)+0;
    printf("%d\n", random1);
    if (0&lt;=a&lt;=17)
    printf("good\n");
    else
    printf("bad\n");
return 0;
}
</code></pre>
<p>This only prints ‘good’ when ‘a’ isnt in range.</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="asdklfjsad" data-post="1" data-topic="210571">
<div class="title">
<div class="quote-controls"></div>
 asdklfjsad:</div>
<blockquote>
<p>This only prints ‘good’ when ‘a’ isnt in range.</p>
</blockquote>
</aside>
<p>This actually <em>always</em> prints “good”, and it has nothing to do with the random number. The reason is that you combine expressions in the <code>if</code> condition:</p>
<aside class="quote no-group" data-username="asdklfjsad" data-post="1" data-topic="210571">
<div class="title">
<div class="quote-controls"></div>
 asdklfjsad:</div>
<blockquote>
<p><code>    if (0&lt;=a&lt;=17)</code></p>
</blockquote>
</aside>
<p>I’ll add parentheses to show what actually happens here:</p>
<pre><code class="lang-auto">if ((0 &lt;= a) &lt;= 17)
</code></pre>
<p>The result of <code>0 &lt;= a</code> is always either 0 (false) or 1 (true). And both 0 and 1 are less than or equal to 17. :slightly_smiling_face:</p>
<p>If you want to make sure that a is in the range from 0 to 17 (inclusive) you need to check both conditions separately and combine them with a <code>&amp;&amp;</code> (logical and) operator.</p>

**Comment by asdklfjsad** [comment]
<p>I understand! thanks</p>
